lib/kv/manager/src/layout/contiguous.rs

```diff
@@ -0,0 +1,528 @@
+// SPDX-FileCopyrightText: Copyright (c) 2024-2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! ### Contiguous Memory Layout
+//!
+//! This module provides the implementation for a block-continuous memory layout
+//! of KV blocks.
+//!
+//! A single allocation of memory, minimally of size [nblocks, nlayers, page_size, inner_dim],
+//! is required to
+
+use super::*;
+
+use crate::storage::{Storage, StorageAllocator};
+
+/// Internal struct to hold calculated layout dimensions specific to FullyContiguous.
+// Module-level, but only used internally by FullyContiguous
+#[derive(Debug, Clone)]
+struct FullyContiguousConfig {
+    inner: LayoutConfig,
+    memory_region_size: usize,
+    layer_stride_in_bytes: usize,
+    natural_block_stride: usize,
+    block_stride_in_bytes: usize, // Aligned if necessary
+    layout_data_bytes: usize,     // Size of the layout data itself (post base offset)
+}
```

Comment:
Would Rust's memory `Layout` be useful? https://doc.rust-lang.org/stable/alloc/alloc/struct.Layout.html